src/chat/chatroom.js

```diff
@@ -58,16 +82,26 @@ export default class Chatroom extends Component {
     });
   }
 
-  getMessagesOnMount() {
-    this.props.socket.on('updateMessagesState', updatedChatRoom => {
-      const messages = updatedChatRoom ? updatedChatRoom.messages : [];
-      this.setState({ messageList: messages });
+  listenForGeoChange() {
+    this.watchID = navigator.geolocation.watchPosition(position => {
+      const coordStr = this.createChatRoomId(position.coords);
+      if (coordStr !== this.state.location) {
+        log('location changed, return to map');
```

Comment:
Yo! Try using a ternary operator to render an Alert component based on some state variable set here.